internal/wrappers/byor-http.go

```diff
@@ -0,0 +1,78 @@
+package wrappers
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	errorConstants "github.com/checkmarx/ast-cli/internal/constants/errors"
+	"github.com/checkmarx/ast-cli/internal/logger"
+	commonParams "github.com/checkmarx/ast-cli/internal/params"
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+)
+
+const (
+	importsPath       = "/imports"
+	successfulMessage = "The SARIF results were successfully imported into project %s importID: %s"
+)
+
+type ByorHTTPWrapper struct {
+	path          string
+	clientTimeout uint
+}
+
+func NewByorHTTPWrapper(path string) ByorWrapper {
+	return &ByorHTTPWrapper{
+		path:          path,
+		clientTimeout: viper.GetUint(commonParams.ClientTimeoutKey),
+	}
+}
+func (b *ByorHTTPWrapper) Import(projectID, uploadURL string) (string, error) {
+	req := CreateImportsRequest{
+		ProjectID: projectID,
+		UploadURL: uploadURL,
+	}
+
+	jsonBytes, _ := json.Marshal(req)
+	resp, err := SendHTTPRequestWithJSONContentType(http.MethodPost, b.path+importsPath, bytes.NewBuffer(jsonBytes), true, b.clientTimeout)
+	if err != nil {
+		return "", err
+	}
+	defer func() {
+		_ = resp.Body.Close()
+	}()
+	decoder := json.NewDecoder(resp.Body)
+	switch resp.StatusCode {
+	case http.StatusForbidden:
+		return "", getError(decoder, errorConstants.StatusForbidden)
+	case http.StatusUnauthorized:
+		return "", getError(decoder, errorConstants.StatusUnauthorized)
+	case http.StatusInternalServerError:
+		return "", getError(decoder, errorConstants.StatusInternalServerError)
+	case http.StatusOK:
+		model := CreateImportsResponse{}
+		err = decoder.Decode(&model)
+		if err != nil {
+			return model.ImportID, errors.Errorf("Parsing upload model failed - %s", err.Error())
+		}
+		logger.Printf(successfulMessage, projectID, model.ImportID)
+		return model.ImportID, nil
+	case http.StatusNotFound:
+		fmt.Printf("status code: %d b.path+importsPath: %s request: %s\n", http.StatusNotFound, b.path+importsPath, req)
+		return "", errors.Errorf("Import was not successful for project ID %s", projectID)
```

Comment:
when are we getting this 404? why should we get it? is it sever error of project not exits? if project not exsits, let's response this?

Reply:
I think I got this one when there was some temporary issue with the byor service and it returned 404.
It sends the upoad url it gets from the the Upload service, which was generated successfully.

Reply:
Removed

Reply:
please commit